File: 383. Ransom Note.go
package main

import (
	"fmt"
)

func canConstruct(ransomNote string, magazine string) bool {

	magFreq := map[byte]int{}

	for i := 0; i < len(magazine); i++ {
		if _, didFind := magFreq[magazine[i]]; didFind {
			magFreq[magazine[i]]++
		} else {
			magFreq[magazine[i]] = 1
		}
	}

	for i := 0; i < len(ransomNote); i++ {
		if _, didFind := magFreq[ransomNote[i]]; didFind {
			if magFreq[ransomNote[i]] > 0 {
				magFreq[ransomNote[i]]--
			} else {
				return false
			}
		} else {
			return false
		}
	}

	return true
}

func main() {
	fmt.Println("Creit Sussie")
}

File: 383. Ransom Note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        
        mMap = defaultdict(int)

        for ch in magazine:  mMap[ch] += 1

        for ch in ransomNote:
            if ch not in mMap or mMap[ch] <= 0: return False
            mMap[ch] -= 1
        
        return True
        

